src/BuildingRegistry/Building/Building.cs

```diff
@@ -52,6 +52,88 @@ public sealed partial class Building : AggregateRootEntity, ISnapshotable
+
+            foreach (var building in buildingsToMerge)
+            {
+                if (building.BuildingStatus != BuildingStatus.Realized)
+                    throw new BuildingToMergeHasInvalidStatusException();
+
+                if (building.BuildingGeometry.Method != BuildingGeometryMethod.MeasuredByGrb)
+                    throw new BuildingToMergeHasInvalidGeometryMethodException();
+            }
+
+            var buildingGeometry = WKBReaderFactory.Create().Read(extendedWkbGeometry);
+            GuardPolygon(buildingGeometry);
+
+            var newBuilding = buildingFactory.Create();
+            newBuilding.ApplyChange(
+                new BuildingMergerWasRealized(
+                    buildingPersistentLocalId,
+                    extendedWkbGeometry,
+                    buildingsToMerge.Select(x => x.BuildingPersistentLocalId)));
+
+            foreach (var buildingToMerge in buildingsToMerge)
+            {
+                var buildingUnitsToTransfer =
+                    buildingToMerge._buildingUnits.PlannedBuildingUnits().Concat(buildingToMerge._buildingUnits.RealizedBuildingUnits());
+
+                foreach (var buildingUnit in buildingUnitsToTransfer.Where(x => x.Function != BuildingUnitFunction.Common))
+                {
+                    var geometryMethod = buildingUnit.BuildingUnitPosition.GeometryMethod;
+
+                    if (!newBuilding.BuildingGeometry.Contains(buildingUnit.BuildingUnitPosition.Geometry))
+                    {
+                        geometryMethod = BuildingUnitPositionGeometryMethod.DerivedFromObject;
+                    }
+
+                    var geometryPosition = buildingUnit.BuildingUnitPosition.Geometry;
+
+                    if (geometryMethod == BuildingUnitPositionGeometryMethod.DerivedFromObject)
+                    {
+                        geometryPosition = newBuilding.BuildingGeometry.Center;
+                    }
+
+                    newBuilding.ApplyChange(new BuildingUnitWasTransferred(
+                        buildingPersistentLocalId,
+                        buildingUnit,
+                        buildingToMerge.BuildingPersistentLocalId,
+                        new BuildingUnitPosition(geometryPosition, geometryMethod)));
+
+                    newBuilding.EnsureCommonBuildingUnit(addCommonBuildingUnit);
+                }
+
+                if (buildingToMerge._buildingUnits.HasCommonBuildingUnit())
+                {
+                    var commonBuildingUnitToMerge = buildingToMerge._buildingUnits.CommonBuildingUnit();
+                    var newCommonBuildingUnit = newBuilding._buildingUnits.CommonBuildingUnit();
+
+                    foreach (var addressToTransfer in commonBuildingUnitToMerge.AddressPersistentLocalIds)
+                    {
+                        newBuilding._buildingUnits.CommonBuildingUnit();
+                        newBuilding.ApplyChange(new BuildingUnitAddressWasAttachedV2(
```

Comment:
Should we also perform an explicit detach? Or is this unnecessary as the common building unit will never be used again?